### DISCUSSION TITLE
Migrating large Hugo site curiosity

### DISCUSSION THREAD

**Discussion by shah** [lumeland/lume#99] [Q&A] [answered]
Hi, we have a large site (over 100k pages which we build many times a day) that Hugo builds within a couple of minutes. We're very happy with Hugo performance and base functionality but we add on many Deno and shell scripts to do a lot of pre and post processing in Hugo which seems possible with Lume's plugins. 

For example, all our pages' markdown are generated from SQL data but all that work happens outside of Hugo in Deno scripts because Hugo has no hooks capability or ability to generate pages from PostgreSQL. 

I know this question would be difficult to answer but I was curious what the HTML build performance might be for over 100k markdown files in Lume and would love to hear some general opinions and wild guesses. :-)

This is a fantastic library, great work!

We would happy to be a trial site for a large Lume project if there's interest.

**Comment by oscarotero** [comment, marked as answer]
Hi, @shah 
Thanks for your interest in Lume. In terms of performance, I think Hugo is impossible to beat (any compiled language like Go will be always faster than a scripting language such javascript). Maybe using some wasm/rust libraries that can work on Deno could increase the performance.

I did some benchmarking some time ago here: https://github.com/lumeland/benchmark and Lume takes (more or less) the double of time to build the same site. If you think the ability of Lume to be extended with hooks and integrated with databases worths the performance reduction, I'd give it a try.

I'm open to PR and suggestions to improve performance.

**Reply by shah** [reply]
Thanks for the quick reply @oscarotero! Being _just_ twice as slow in an interpreted language with plugins capability seems to be worth it so I'll give it a shot and see how it goes. You're right that we could write Rust-based Deno plugins once we figure out where improvements may be useful.

That benchmark page was quite helpful, thanks for sharing.
